tfexec/options.go

```diff
@@ -4,6 +4,16 @@ import (
 	"encoding/json"
 )
 
+// AddressOption represents an address.
```

Comment:
There may be opinions regarding how this is named, or even implemented. `terraform state show` is documented as taking zero or more addresses as options.  As such I added an AddressOption, which ends up getting applied (for `StateList`) as positional arguments after all the other options.

I'm more than happy to change how this is implemented if there are other suggestions.